src/path.js

```diff
@@ -16,6 +16,283 @@ function Path() {
+                previousCommand.x === firstCommand.x &&
+                previousCommand.y === firstCommand.y
+            ) {
+                subpath.shift();
+                subpath[0].type = 'M';
+            }
+
+            if (i + 1 < commands.length) {
+                subpaths.push([]);
+            }
+        } else if (cmd.type === 'L') {
+            // remove lines that lead to the same position as the previous command
+            if (previousCommand.x === cmd.x && previousCommand.y === cmd.y) {
+                subpath.pop();
+            }
+        }
+    }
+    commands = [].concat.apply([], subpaths); // flatten again
+    return commands;
+}
+
+/**
+ * Returns options merged with the default options for parsing SVG data
+ * @param {object} options (optional)
+ */
+function defaultSVGParsingOptions(options) {
+    const defaultOptions = {
+        decimalPlaces: 2,
+        optimize: true,
+        flipY: true,
+        scale: 1,
+        x: 0,
+        y: 0
+    };
+    const newOptions = Object.assign({}, defaultOptions, options);
+    return newOptions;
+}
+
+/**
+ * Returns options merged with the default options for outputting SVG data
+ * @param {object} options (optional)
+ */
+function defaultSVGOutputOptions(options) {
+    // accept number for backwards compatibility
+    // and in that case set flipY to false
+    if (parseInt(options) === options) {
+        options = { decimalPlaces: options, flipY: false };
+    }
+    const defaultOptions = {
+        decimalPlaces: 2,
+        optimize: true,
+        flipY: true
+    };
+    const newOptions = Object.assign({}, defaultOptions, options);
+    return newOptions;
+}
+
+/**
+ * Sets the path data from an SVG path element or path notation
+ * @param  {string|SVGPathElement}
+ * @param  {object}
+ */
+Path.prototype.fromSVG = function(pathData, options = {}) {
+    if (typeof SVGPathElement !== 'undefined' && pathData instanceof SVGPathElement) {
+        pathData = pathData.getAttribute('d');
+    }
+
+    // set/merge default options
+    options = defaultSVGParsingOptions(options);
+
+    this.commands = [];
+
+    const number = '0123456789';
+    const supportedCommands = 'MmLlQqCcZzHhVv';
+    const unsupportedCommands = 'SsTtAa';
+    const sign = '-+';
+
+    let command = {};
+    let buffer = [''];
+
+    let isUnexpected = false;
+
+    function parseBuffer(buffer) {
+        return buffer.filter(b => b.length).map(b => {
+            let float = parseFloat(b);
+            if (options.decimalPlaces || options.decimalPlaces === 0) {
+                float = roundDecimal(float, options.decimalPlaces);
+            }
+            return float;
+        });
+    }
+
+    function makeRelative(buffer) {
+        if (!this.commands.length) {
+            return buffer;
+        }
+        const lastCommand = this.commands[this.commands.length - 1];
+        for (let i = 0; i < buffer.length; i++) {
+            buffer[i] += lastCommand[i % 2 ? 'y' : 'x'];
```

Comment:
Just a small optimization, ``x % 2`` can always be rewritten as ``x & 1`` which runs faster, not that big of a deal, but processors are a lot slower at modulus (due to the division).

Reply:
Interesting! That's good to know. We could probably optimize performance further by using a generator function and yield, but our current build process doesn't transpile that. I'll add that as a todo comment.

Reply:
Chromium browsers seem to do some optimizations that make the difference marginal (and sometimes modulus runs even a tiny bit better than bitwise), but the story is different in Firefox. I had differences of up to ~15%:

https://jsbench.me/2zlehfyyi2/1

![image](https://user-images.githubusercontent.com/13076806/221000244-14b2d60d-9c64-4e48-89a9-8cbb6ada108b.png)

Reply:
I confirm that, on chromium-based browser get almost identical result for both `%` and `&`.

The PR is (re)open so you can merge it (no issue for me :+1: ).

Let's just not forget that `premature optimization is the root of all evil` ;)

Reply:
@yne the reason why it's the same on chrome is that chrome rewrites the code to actually be ``x & 1`` when it sees ``x % 2``.